### ISSUE TITLE
[Feature request] Notify on failure/completion in single task view

### ISSUE THREAD

**Issue by reuben** [taskcluster/taskcluster#2255]
When viewing a single task (e.g. as a result of editing a task), I can't find any way to get notified on task failure/completion like it's possible in the group view.

**Comment by helfi92** [comment]
That's a good idea. A good place for this would probably be in the top right section of that view.

![Screen Shot 2020-01-16 at 9 13 30 AM](https://user-images.githubusercontent.com/3766511/72532096-c5740980-3840-11ea-99c3-dc64380a110f.png)

**Comment by virgilt** [comment]
@helfi92 Can I work on this?

**Comment by helfi92** [comment]
Go for it @virgilt. We do something similar in the task group view https://community-tc.services.mozilla.com/tasks/groups/c727T42WRGmg060OtQkC-g so you could take a look to see how it is implemented and possibly re-use some of the code. Let me know if you have any questions. Happy to help.

**Comment by virgilt** [comment]
@helfi92 Ok, thanks for the tip!

**Comment by virgilt** [comment]
@helfi92 I am having some problems setting up the postgres server with docker, following the instructions here: https://github.com/taskcluster/taskcluster/blob/master/dev-docs/development-process.md

I am not sure how to set the READ_DB_URL, WRITE_DB_URL, or TASKCLUSTER_ROOT_URL.

**Comment by helfi92** [comment]
To run the db, you could do:

```bash
docker run -ti -p 127.0.0.1:5432:5432  --rm postgres:11
```

To run the UI, you can do:

```
export READ_DB_URL=postgresql://postgres@localhost/postgres
export WRITE_DB_URL=postgresql://postgres@localhost/postgres
export TASKCLUSTER_ROOT_URL="https://community-tc.services.mozilla.com"
yarn
yarn start
```

Does that help?

**Comment by virgilt** [comment]
@helfi92 That did the trick. I was not exporting READ_DB_URL and WRITE_DB_URL. I followed the rest of the instructions fine. Thanks!

**Comment by virgilt** [comment]
@helfi92 Just letting you know that I am still working on this feature. I have the notification button functioning. I am working out tracking the task right now.

**Comment by helfi92** [comment]
Many thanks for the update. Great to hear that there's been some progress :)

**Comment by virgilt** [comment]
@helfi92 Yes making progress. I am debugging a props error, but I have located the cause, now just to figure out the why.

**Comment by virgilt** [comment]
@helfi92 If I want to test the task completion notification how would you suggest I do it? Pass the expected value manually?

**Comment by helfi92** [comment]
I thought I had replied to this message. Sorry for the delay!

There's a lot of jobs shown in https://treeherder.mozilla.org/#/jobs?repo=try that you can try and inspect if they show the notification.

1. Click any letter button in the right hand side of that view
<img width="723" alt="Screen Shot 2020-06-08 at 8 55 48 AM" src="https://user-images.githubusercontent.com/3766511/84032802-dcd59f80-a965-11ea-9dce-50f5b8f581d9.png">

2. Click on the link to go to the task view
<img width="368" alt="Screen Shot 2020-06-08 at 8 56 39 AM" src="https://user-images.githubusercontent.com/3766511/84032965-28884900-a966-11ea-8aa9-6b814912352f.png">

Hope that helps!

**Comment by virgilt** [comment]
@helfi92 I tried using notifications for the group view to see what exactly how the notifications work, but each time I refresh the notifications are removed. I clicked the watch button but it doesn't seem to recall that I want notifications despite being logged in.

**Comment by helfi92** [comment]
Here's what I tried and worked for me:

1. Navigate to https://firefox-ci-tc.services.mozilla.com/tasks/groups/bqDefKflQvmTxUEkqZemJg
2. Click on the notifications button
3. Select which notifications to be notified on
4. Click on the SAVE button
5. Refresh the page
6. Click on the notifications button
7. See that whatever you picked from step 3 is still selected.

Are you seeing a different behavior? I tried this on both Firefox and Chrome and they both worked for me.

**Comment by joeyouss** [comment]
Hi!
Is this issue still open to take up? I am a beginner and found out this issue to be beginner-friendly. I would like to work on it if it is open.

**Comment by djmitche** [comment]
Hi @joeyouss -- sure!  You can see some of the first steps that @virgilt took, including getting the development environment up and running, and @helfi92 had some advice on where to look for inspiration (the task-group viewer).

**Comment by joeyouss** [comment]
@djmitche thankyou ! I am a complete beginner so would be needing a little help 
.also, do you have any chat channel i could connect with the mentors for my doubts ?

**Comment by djmitche** [comment]
Sure, there's #taskcluster on https://chat.mozilla.org

**Comment by joeyouss** [comment]
@djmitche thank you!
